app/controllers/controllers.js

```diff
@@ -201,58 +201,74 @@ exports.showAllTeams = (req, res) => {
 
 // Handles teammate evaluation
 exports.evaluateTeammate = async (req, res) => {
-  if (req.session.user) {
-    var teammateID = req.params.id;
-    var reviewerID = req.session.user.id;
+  // Check if user is logged in (session should exist)
+  if (!req.session.user || !req.session.user.id) {
+    return res.redirect("/login"); // Redirect to login page if user is not logged in
+  }
+
+  const teammateID = req.params.id;
+  const reviewerID = req.session.user.id;
 
-    var sql =
-      `SELECT * FROM evaluations WHERE teammateID = ${teammateID} ` +
-      `AND reviewerID = ${reviewerID}`;
+  console.log("Teammate ID:", teammateID);
+  console.log("Reviewer ID:", reviewerID);
 
+  if (!teammateID || !reviewerID) {
+    return res.status(400).send("Invalid IDs provided.");
+  }
+
+  try {
+    // Query to check if the current reviewer has already evaluated the teammate
+    let sql = `
+      SELECT 
+        TypeOfEval, score, comments
+      FROM evaluations 
+      WHERE teammateID = ${teammateID} AND reviewerID = ${reviewerID}
+    `;
     const result = await db.query(sql);
 
-    // Checks if an evaluation has been submitted already
-    if (result[0].length > 0) {
-      res.render("ViewEvaluation.ejs", { review: result[0][0] });
-    } else {
+    // If no evaluation exists, fetch teammate details and render the evaluation page
+    if (result[0].length === 0) {
       sql = `SELECT ID, Username FROM students WHERE ID = ${teammateID}`;
-      db.query(sql)
-        .then((teammate) => {
-          res.render("Evaluation.ejs", { teammate: teammate[0][0] });
-        })
-        .catch((err) => console.log(err));
+      const teammate = await db.query(sql);
+      
+      res.render("Evaluation.ejs", { teammate: teammate[0][0] });
+    } else {
+      // Aggregating evaluation categories into a single object
+      const review = {
+        Cooperation: null,
+        WorkEthic: null,
+        PracticalContribution: null,
+        ConceptualContribution: null,
+      };
+
+      // Assign each evaluation category to the respective key
+      result[0].forEach(evaluation => {
+        review[evaluation.TypeOfEval] = {
+          score: evaluation.score,
+          comments: evaluation.comments
+        };
+      });
+
+      // Render the ViewEvaluation page with the structured review data
+      res.render("ViewEvaluation.ejs", { review });
     }
-  } else {
-    res.sendStatus(200);
+  } catch (err) {
+    console.log(err);
+    res.status(500).send("Error occurred while processing the evaluation.");
   }
 };
 
-exports.submitEvaluation = (req, res) => {
-  if (req.session.user) {
-    const { teammateID, TypeOfEval, score, comments } = req.body;
-    const reviewerID = req.session.user.id;
```

Comment:
Ensure that you validate and sanitize the user input to avoid common security risks.

Validation: Ensure that inputs like teammateID, TypeOfEval, score, and comments are in the correct format and within the expected range.
Sanitization: For fields like comments, use a sanitization library like DOMPurify or validator to escape potentially dangerous HTML.